spec/pubid_iso/identifier_parsing_spec.rb

```diff
@@ -406,6 +406,14 @@ module Pubid::Iso
       end
     end
 
+    context "ISO 10303-1047-20-21:2014 ED3" do
+      let(:original) { "ISO 10303-1047-20-21:2014 ED3" }
```

Comment:
@ronaldtse The issue was caused when the `part` contains dashes e.g `13584-20` it converts it into an array that is why we were getting names with arrays so the fix is to convert them back to string during transformation. 
There is a [test case for part without dashes](https://github.com/metanorma/pubid-iso/blob/865e6c2d36a49300fd986e75c27716773baf4962/spec/pubid_iso/identifier_parsing_spec.rb#L404) so I added another one for `part with dashes`

Reply:
This is funny because `13584-20` is a violation in the data -- we should never actually process them.

But it is true that ISO/IEC parts can have subparts (i.e. 'xxxx-11-3').

@mico can you check this PR?

Reply:
@ronaldtse do you think is it correct to keep all parts in "part" attribute as one string?
For identifier like `IEC/DPAS 63086-3-1` "part" attribute will contain "3-1" and for 
`ISO 10303-1047-20-21` (fake identifier) it will be "1047-20-21".

Reply:
Yeah I think this is the correct approach. Thanks!!